node/subsystem-util/src/rolling_session_window.rs

```diff
@@ -212,7 +212,12 @@ impl RollingSessionWindow {
 	}
 }
 
-async fn get_session_index_for_head(
+// Returns the session index expected at any child of the `parent` block.
+//
+// Note: This internal `fn` is a duplicate of the one exported by `subsystem-util`,
+// that is because `RollingSessionWindow` already implements it's own caching mechanism.
+// It would not make sense to have 2 layers of caching.
+async fn get_session_index_for_child(
```

Comment:
I don't understand why we would not try to use the shared cache to populate our own rolling window?

Reply:
That would imply filling another cache in `RuntimeInfo` that is if we would construct and keep it around.

Reply:
The cache is there anyways :shrug:

Reply:
We have a "remote" cache in the runtime api subsystem. `RollingSessionWindow` and `RuntimeInfo` both implement local caches. Does not really make sense to pull in one into the other.